src/main/java/com/example/wbdvsp20bazaarjavaserver/controllers/UserController.java

```diff
@@ -65,4 +65,28 @@ public User profile(HttpSession session) {
         return profile;
     }
 
+    @PutMapping("/api/users/{uid}")
+    public int updateUser(HttpSession session,
+        @PathVariable int uid,
+        @RequestBody User updatedUser) {
+
+        // Allow updates only if the session User Id matches the Id of User being updated
+        if (((User)session.getAttribute("profile")).getId() == updatedUser.getId()) {
+            int updateStatus = this.service.updateUser(uid, updatedUser);
+            if (updateStatus == 1) {
+                // Update Session profile if the update was successful.
+                session.setAttribute("profile", updatedUser);
+            }
+            return updateStatus;
+        }
+        else { // Session User id does not match the id of the user being updated.
+            return 0;
+        }
+    }
+
+    @DeleteMapping("/api/users/{uid}")
+    public int deleteUser(@PathVariable("uid") int uid) {
+        return service.deleteUser(uid);
```

Comment:
This is what I initially had to check session id and id of user being deleted, but help me what's wrong..
```
@DeleteMapping("/api/users/{uid}")
    public int deleteUser(HttpSession session, @PathVariable("uid") int uid) {
        // Allow updates only if the session User Id matches the Id of User being deleted
        if (((User)session.getAttribute("profile")).getId() == uid) {
            return service.deleteUser(uid);
        }
        else { // Session User id does not match the id of the user being deleted.
            return 0;
        }
    }

Reply:
Resolved. The code you had was good. Credentials for delete request on the client side was not provided (HTTPSession had no session to refer to when it received the request).